OpenOversight/app/models/database.py

```diff
@@ -34,7 +37,66 @@
 
 db = SQLAlchemy()
 jwt = JsonWebToken(SIGNATURE_ALGORITHM)
-BaseModel: DeclarativeMeta = db.Model
+Base: DeclarativeMeta = db.Model
+
+
+class BaseModel(Base):
+    __abstract__ = True
+
+    EXCLUDED = [
+        "approved_at",
+        "approved_by",
+        "confirmed_at",
+        "confirmed_by",
+        "created_at",
+        "created_by",
+        "disabled_at",
+        "disabled_by",
+        "password_hash",
+        "last_updated_at",
+        "last_updated_by",
+    ]
+
+    def __repr__(self) -> str:
+        ret_str = f"<{self.__class__.__name__} ("
+        for column in inspect(self).mapper.column_attrs:
+            if column.key in self.EXCLUDED or column.key.startswith("_"):
+                continue
+
+            if ret_str[-1] != "(":
+                ret_str += " : "
+
+            value = getattr(self, column.key)
+            if isinstance(value, (date, datetime)):
+                ret_str += f"{column.key}: {value.isoformat()}"
+            elif isinstance(value, date):
+                ret_str += f'{column.key}: {value.strftime("%Y-%m-%d")}'
+            elif isinstance(value, dt_time):
+                ret_str += f'{column.key}: {value.strftime("%I:%M %p")}'
+            else:
+                ret_str += f"{column.key}: {value}"
+
+        return ret_str + ")>"
+
+    def to_dict(self) -> dict:
```

Comment:
Adds a dictionary function that can be used for serialization in data API ticket: https://github.com/lucyparsons/OpenOversight/issues/385